### ISSUE TITLE
`Parse-SDK-iOS-OSX/index.lock` from git on `carthage update`

### ISSUE THREAD

**Issue by CiraciNicolo** [Carthage/Carthage#2201]
Carthage check out the same packages twice at different tags.

* `carthage version`: 0.25.0 
* `xcodebuild -version`: Xcode 9.0 Build version 9A235 
* Are you using `--no-build`?  No
* Are you using `--no-use-binaries`?  Yes
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No

**Cartfile**
```
github "parse-community/Parse-SDK-iOS-OSX"
```

**Carthage Output**
```
*** Checking out Parse-SDK-iOS-OSX at "1.15.3"
*** Checking out Parse-SDK-iOS-OSX at "1.15.0"
A shell task (/usr/bin/env git checkout --quiet --force 1.15.0 (launched in /Users/user/Library/Caches/org.carthage.CarthageKit/dependencies/Parse-SDK-iOS-OSX)) failed with exit code 128:
fatal: Unable to create '/Users/user/Library/Caches/org.carthage.CarthageKit/dependencies/Parse-SDK-iOS-OSX/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
```

**Comment by jdhealy** [comment]
Wasn’t able to repro this with Carthage 0.27.0…

Could anybody hitting this issue regularly and reproducibly try with 0.27.0 and report back?

**Comment by jdhealy** [comment]
Closing due to inactivity.

**Comment by CiraciNicolo** [comment]
Still present in 0.27 with plain `carthage update`

**Comment by blender** [comment]
Are you running multiple Carthage commands in parallel?

**Comment by Renleilei1992** [comment]
@blender nope, just one process. use `Carthage update` command in iterm2.  (also restart my Mac, only open 1 iterm window,  `ps -ef | grep git` output nothing),  thank you for the response. : )

**Comment by tomieq** [comment]
I run into the same problem and I think it is connected with dependencies. My configuration worked for me until I had to switch from github to forked repository hosted locally.

In my project I have depencency to AlamofireObjectMapper. In my Cartfile I have both Alamofire and AlamofireObjectMapper. AlamofireObjectMapper  internally has dependencies to Alamofire hosted by github.
When I switched to entry git "local.repo/Alamofire" and git "local.repo/AlamofireObjectMapper" carthage tries do download both Alamofire from local repo(as in the Cartfile) and Alamofire from github(as defined in AlamofireObjectMapper  dependency).
